src/checkers/inference/solver/backend/encoder/AbstractConstraintEncoderFactory.java

```diff
@@ -22,8 +23,11 @@
      */
     protected final ConstraintVerifier verifier;
 
-    public AbstractConstraintEncoderFactory(Lattice lattice, ConstraintVerifier verifier) {
+    protected final FormatTranslatorT formatTranslator;
+
+    public AbstractConstraintEncoderFactory(Lattice lattice, ConstraintVerifier verifier, FormatTranslatorT formatTranslator) {
         this.lattice = lattice;
         this.verifier = verifier;
+        this.formatTranslator = formatTranslator;
```

Comment:
Where is this field used?

Reply:
Here:https://github.com/opprop/checker-framework-inference/blob/276e57bd13cdc35650e9b1434743fe405e46c1a2/src/checkers/inference/solver/backend/maxsat/encoder/MaxSATConstraintEncoderFactory.java#L57
Since `ImplicationConstraint` is a composite `Constraint`, it needs delegating to `FormatTranslator` to encode ingredient `Constraint`s.

Reply:
`Z3` backend also needs `FormatTranslator` in some encoder, so I raise this up to `AbstractConstraintEncoderFactory`